scripts/remove_after_use/fix_unclaimed_records.py

```diff
@@ -0,0 +1,35 @@
+import sys
+import logging
+from website.app import setup_django
+
+setup_django()
+from osf.models import OSFUser
+logger = logging.getLogger(__name__)
+
+
+def add_missing_unclaimed_record(user, node, dry_run):
+    # Get referrer from logs
+    for log in node.logs.filter(action='contributor_added').order_by('date'):
+        if user._id in log.params['contributors']:
+            if log.user_id:
+                referrer = OSFUser.objects.get(id=log.user_id)
+                user.add_unclaimed_record(node, referrer, user.fullname)
```

Comment:
Fixing the unicode issue locally, found that the referrer isn't guaranteed to have permission here (i.e. if they're no longer on the node, if the node is deleted, etc.).

Reply:
The problem is using the add_unclaimed_record method. It does validation of the referrer and some other stuff. Since these should be created based on node logs they should be ensured valid.

In my somewhat related fix, I built the records manually to avoid extra work:
https://github.com/CenterForOpenScience/osf.io/blame/88097752989f79652e798982e0d4e59d43bf13e7/scripts/fix_registration_unclaimed_records.py#L52